eng/pipelines/templates/steps/archetype-sdk-docs.yml

```diff
@@ -14,138 +14,19 @@ steps:
       BuildSHA: $(Build.SourceVersion)
       RepoId: 'Azure/azure-sdk-for-net'
   - pwsh: |
-      # Download and Extract or restore Packages required for Doc Generation
-      Write-Host "Download and Extract mdoc to Build.BinariesDirectory/mdoc"
-      Invoke-WebRequest -MaximumRetryCount 10 -Uri "https://github.com/mono/api-doc-tools/releases/download/mdoc-5.7.4.9/mdoc-5.7.4.9.zip" `
-      -OutFile "mdoc.zip" | Wait-Process; Expand-Archive -Path "mdoc.zip" -DestinationPath "./mdoc/"
-
-      Write-Host "Download and Extract docfx to Build.BinariesDirectory/docfx"
-      Invoke-WebRequest -MaximumRetryCount 10 -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
-      -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "./docfx/"
-
-      Write-Host "Restore eng/docgeneration/assets/docgen.csproj, to get ECMA2Yml and popimport"
-      dotnet restore "$(Build.SourcesDirectory)/eng/docgeneration/assets/docgen.csproj" /p:BuildBinariesDirectory=$(Build.BinariesDirectory)
-    workingDirectory: $(Build.BinariesDirectory)
+      . $(Build.SourcesDirectory)/doc/ApiDocGeneration/Generate-Api-Docs.ps1
```

Comment:
Instead of "." which inputs the functions I would just make this a callable script with parameters and simply call it.

Reply:
Was trying to keep all doc generation in a single script since the aim was for it to be easily reproduced locally.

Reply:
Separated into different script.